test/index.test.js

```diff
@@ -1,10 +1,44 @@
-// You can import your modules
-// const index = require('../index')
+const { Application } = require('probot')
+// Requiring our app implementation
+const plugin = require('..')
```

Comment:
I don't think we should use plugin here. I'd prefer to rename this to app and rename app to probot.

Reply:
🤔 Hm, but the class is called Application, so that would look like this:

```
const { Application } = require('probot')
const probot = new Application()
const app = require('..')
...
probot.load(app)
```

We do tend to call these plugins elsewhere (like the 3 default plugins for stats, sentry, and the default website). But I see the confusion since we do `module.exports = app => {...}`. I'm inclined to leave it as plugin and maybe explain that reasoning a little better in the docs?

Reply:
I'm having flashbacks to https://github.com/probot/probot/pull/210 but I think we should be very certain about our language choices before merging this, since everyone will use that language once we do it.

I also don't think that code snippet is all that horrible sounding, but I'm open to other thoughts.

Reply:
I agree with @tcbyrd that that code snippet is confusing - I'd expect `app` to be an instance of `Application`, just based on the naming.

However, I agree that `plugin` is 👎, especially because of that issue @hiimbex was kind enough to link to. Some other ideas:

* `applicationFunction` (which is actually the TS name)
* `appFunction`
* `myApp`
* `myProbotApp`
* Fill this with a camel-cased version of their app name, as part of the `create-probot-app` script

Reply:
I think all of these are good options, but the idea of mimicking the actual TS name sounds the best to me. But I don't have strong opinion on this except to _not_ use plugin

Reply:
I think what we're struggling with here is the default export of the template doesn't have an app name. What we should probably be encouraging is using the name of the app instead of "plugin".

So for TODO:
```
const { Application } = require('probot')
const app = new Application()
const todo = require('..')
app.load(todo)
```

or WIP:
```
const { Application } = require('probot')
const app = new Application()
const wip = require('..')
app.load(wip)
```

I like using `applicationFunction` because it's the best description of the general term, but maybe we could leave a comment that it could be whatever your name your app?

Reply:
@tcbyrd, thanks for finding those examples. Given that we name the variable after our apps, it sounds like this idea would be the best, but also the most effort to implement:

> Fill this with a `camel-cased` version of their app name, as part of the `create-probot-app` script

Reply:
Et voilà: https://github.com/probot/create-probot-app/pull/30

Reply:
One concern of this approach is that it kind of breaks if folks name their probot app just "app" or any other reserved word, but not sure if that is a huuuge problem.

Another option is to use `myProbotApp` instead of `plugin`?